rules/rules-overridden-azure/technology-usage/apm.windup.xml

```diff
@@ -0,0 +1,115 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ruleset id="apm"
+         xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            Identify Application Performance Management dependencies.
+            This ruleset overrides the rules/rules-reviewed/technology-usage/apm.windup.xml ruleset.
+        </description>
+        <dependencies>
+          <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final"/>
+          <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final"/>
+        </dependencies>
+        <targetTechnology id="azure-spring-apps"/>
+        <targetTechnology id="azure-appservice"/>
+        <targetTechnology id="azure-aks"/>
+        <targetTechnology id="azure-container-apps"/>
+        <tag>apm</tag>
+        <overrideRules>true</overrideRules>
+    </metadata>
+    <rules>
+        <rule id="apm-00000">
+            <when>
+                <or>
+                    <project>
+                        <artifact groupId="com.microsoft.azure" artifactId="{*}applicationinsights{*}"/>
+                    </project>
+                    <file filename="applicationinsights{*}.jar"/>
+                </or>
+            </when>
+            <perform>
+                <hint title="Application Performance Management (APM) tool - Application Insights" category-id="mandatory" effort="1">
+                    <message>
+                        To migrate a Java application that uses Application Insights to Azure, you can follow this recommendation:
```

Comment:
If the application already has the App Insights JAR file, do they still need to enable App Insights in the portal page of Azure App Service?

Reply:
> If the application already has the App Insights JAR file, do they still need to enable App Insights in the portal page of Azure App Service?

@brunoborges Yes, it is recommended to use the Azure-managed method to enable App Insights instead of using the JAR file.

Reply:
@KaiqianYang so does this mean that the rule is still `mandatory`? If it is recommended, could that be `optional` instead ?

Reply:
> @KaiqianYang so does this mean that the rule is still `mandatory`? If it is recommended, could that be `optional` instead ?

@agoncal 🤔To use the App Insights jar file, it must be uploaded to a directory in the App Service instance, similar to New Relic. It means either way something mandatory needed to be done. I prefer to leave this hint as it is now because it is recommended and well-documented. WDYT?